src/openforms/appointments/tokens.py

```diff
@@ -0,0 +1,19 @@
+from django.utils import timezone
+
+from openforms.appointments.models import AppointmentInfo
+from openforms.submissions.tokens import SubmissionTokenGenerator
+
+
+class SubmissionAppointmentTokenGenerator(SubmissionTokenGenerator):
+    """
+    Used to generate and check tokens for submission appointments.
+    """
+
+    def get_expiry_days(self, submission):
+        try:
+            return (submission.appointment_info.start_time - timezone.now()).days
+        except (AppointmentInfo.DoesNotExist, TypeError):
+            return super().get_expiry_days(submission)
```

Comment:
I do not understand the logic here, why should these exceptions not cause problems? Why should it fall back to some weird behaviour? This goes against the Zen of Python "Refuse the temptation to guess"